factory/processing/blockProcessorCreator.go

```diff
@@ -1091,6 +1094,8 @@ func (pcf *processComponentsFactory) createVMFactoryShard(
 		GasSchedule:              pcf.gasSchedule,
 		Counter:                  counter,
 		MissingTrieNodesNotifier: notifier,
+		EpochStartTrigger:        epochStartTriggerHandler,
+		RoundHandler:             pcf.coreData.RoundHandler(),
```

Comment:
this is used in consensus as well. So it seems the best component for now.

Reply:
It will not work when you are synchronizing, this one follows the current round, not the one you are executing while syncing. 

For the case where you are not yet in sync with the network, so executing older blocks, you cannot use the RoundHandler from consensus.